File: Spring-backend/Springbackend/src/main/java/com/dravicenne/backend/users/patient/Patient.java
package com.dravicenne.backend.users.patient;

import com.dravicenne.backend.users.User;

import javax.persistence.Entity;
import javax.persistence.Transient;
import java.time.LocalDate;
import java.time.Period;

@Entity(name = "patient")
public class Patient extends User {

    @Transient
    private Integer age;
    private LocalDate DateNaiss;
    private String groupeSang;

    public Patient() {
    }

    public Patient(String nom, String prenom, String email, String password, String confPassword, String status, LocalDate dateNaiss, String groupeSang) {
        super(nom, prenom, email, password, confPassword, status);
        DateNaiss = dateNaiss;
        this.groupeSang = groupeSang;
    }


    public Integer getAge() {
        return Period.between(DateNaiss,LocalDate.now()).getYears();
    }

    public void setAge(Integer age) {
        this.age = age;
    }

    public LocalDate getDateNaiss() {
        return DateNaiss;
    }

    public void setDateNaiss(LocalDate dateNaiss) {
        DateNaiss = dateNaiss;
    }

    public String getGroupeSang() {
        return groupeSang;
    }

    public void setGroupeSang(String groupeSang) {
        this.groupeSang = groupeSang;
    }
}

File: Angular-frontend/src/app/Models/Patient/patient.ts
import { User } from '../User/user';

export class Patient extends User{
  public username: string;
  public groupeSang: string;
  public age: number;
  public dateNaiss: Date;

  // tslint:disable-next-line: max-line-length
  constructor(nom: string, prenom: string, ville: string, phone: string, email: string, password: string, cpassword: string, username: string, groupeSang: string, datenaiss: Date)
  {
    super(nom, prenom, ville, phone, email, password, cpassword);
    this.username = username;
    this.groupeSang = groupeSang;
    this.dateNaiss = datenaiss;
  }

  getUsername(): string
  {
    return this.username;
  }

  getGroupesang(): string
  {
    return this.groupeSang;
  }

  getDatenaiss(): Date
  {
    return this.dateNaiss;
  }
}
